### PR TITLE
Hyp Tests for Two Variables

### PR DESCRIPTION
**Pull request by lilyclements**
This can be reviewed.

There are still many bugs here, however, I have fixed one. We can now do Bernouli, Normal or Poisson tests.
Altered the selector so that the correct data type can be selected.

### CODE REVIEW

File: instat/dlgRegressionSimple.vb [hunk 1]
```diff
@@ -89,6 +89,18 @@ Public Class dlgRegressionSimple
         TestOKEnabled()
     End Sub
 
+    Private Sub DistributionsOffered()
+        If rdoGeneral.Checked Then
+            ucrFamily.SetGLMDistributions()
+        Else
+            If ucrResponse.strCurrDataType = "character" Or ucrResponse.strCurrDataType = "factor" Then
```
Comment (Lunalo) [inline instat/dlgRegressionSimple.vb]: You can do orelse?

File: instat/dlgRegressionSimple.vb [hunk 2]
```diff
@@ -141,15 +147,22 @@ Public Class dlgRegressionSimple
             ucrExplanatory.SetIncludedDataTypes({"integer", "numeric", "character", "factor"})
         ElseIf rdoSpecific.Checked Then
             If ucrFamily.clsCurrDistribution.strNameTag = "Normal" Then
-                ucrResponse.SetIncludedDataTypes({"integer", "numeric"})
+                If ucrFamily.Enabled = False Then
+                    ucrResponse.SetIncludedDataTypes({"integer", "numeric", "character", "factor"})
```
Comment (Lunalo) [inline instat/dlgRegressionSimple.vb]: Why not just exclude Date and Logical? I am not sure

File: instat/dlgRegressionSimple.vb [hunk 3]
```diff
@@ -180,35 +193,77 @@ Public Class dlgRegressionSimple
         Dim clsTFunc As New RFunction
 
         If ucrFamily.clsCurrDistribution.strNameTag = "Poisson" Then
-            If ucrExplanatory.strCurrDataType = "factor" Or ucrExplanatory.strCurrDataType = "character" Then
+            If ucrExplanatory.strCurrDataType = "numeric" Or ucrExplanatory.strCurrDataType = "integer" Then
                 ucrExplanatory.Clear()
             End If
             If ucrResponse.strCurrDataType = "factor" Or ucrResponse.strCurrDataType = "character" Then
```
Comment (Lunalo) [inline instat/dlgRegressionSimple.vb]: Also here, maybe do orelse

File: instat/dlgRegressionSimple.vb [hunk 4]
```diff
@@ -180,35 +193,77 @@ Public Class dlgRegressionSimple
...
+        clsRLengthGrouped.AddParameter("x", ucrResponse.GetVariables().ToScript & "[" & clsRGroup.ToScript & "]")
+        clsRGroup.AddParameter("x", clsROperatorParameter:=clsPoissonOperation)
+        clsPoissonOperation.SetOperation("==")
+        clsPoissonOperation.SetParameter(True, clsRFunc:=ucrExplanatory.GetVariables())
+        clsPoissonOperation.SetParameter(False, strValue:="1") 'in there we have the level, get column name???
+
+
+        clsRLength2.SetRCommand("length")
+        clsRLength2.AddParameter("x", clsRFunctionParameter:=clsRLengthGrouped2)
+        clsRLengthGrouped2.AddParameter("x", ucrResponse.GetVariables().ToScript & "[" & clsRGroup2.ToScript & "]")
```
Comment (Lunalo) [inline instat/dlgRegressionSimple.vb]: ` clsRLengthGrouped2.AddParameter("x", ucrResponse.GetVariables().ToScript & "[" & clsRGroup2.ToScript & "]")`

I havent tested it. but i think it is alredy a string this way 
 clsRLengthGrouped2.AddParameter("x", ucrResponse.GetVariables() & "[" & clsRGroup2.ToScript & "]")

**Comment by lilyclements** [comment]
@Lunalo can you re-review this? It's ready to be merged as far as I can see.

File: instat/ucrDistributions.vb [hunk 5]
```diff
@@ -151,7 +149,22 @@ Public Class ucrDistributions
...
+                            Case "positive integer" ' or integer
+                                If Dist.bPositiveInt Then
+                                    bUse = True
+                                End If
+                            Case "factor" ' or character
+                                If Dist.bTwoLevelFactor Then
+                                    bUse = True
+                                End If
+                        End Select
+                    End If
```
Comment (dannyparsons) [inline instat/ucrDistributions.vb]: We need to rethink distributions generally but happy to leave this for now and come back to this later.

**Pull request merged**